opcua/server/internal_server.py

```diff
@@ -168,19 +171,51 @@ def add_endpoint(self, endpoint):
 
     def get_endpoints(self, params=None, sockname=None):
         self.logger.info("get endpoint")
-        if sockname:
-            # return to client the ip address it has access to
-            edps = []
-            for edp in self.endpoints:
-                edp1 = copy(edp)
-                url = urlparse(edp1.EndpointUrl)
-                url = url._replace(netloc=sockname[0] + ":" + str(sockname[1]))
-                edp1.EndpointUrl = url.geturl()
-                edps.append(edp1)
-            return edps
-        return self.endpoints[:]
+        # return to client the endpoints it has access to
+        edps = pickle.loads(pickle.dumps(self.endpoints))
```

Comment:
what is that? why dumping then loading again?

Reply:
shallow copy() does not copy the endpoints, so either deepcopy() or pickle...